sources/controllers/Register.controller.php

```diff
@@ -1258,4 +1265,42 @@ public function action_registerCheckUsername()
 
 		$context['valid_username'] = !$errors->hasErrors();
 	}
+
+	public function action_agrelang()
```

Comment:
Yeah, I was considering action_agreementlang but it seemed a bit on the longish side.
agreelang looks indeed slightly better than agrelang.